### PR TITLE
Fixes a bug that allowed the multiplication of materials.

### LINKED ISSUE #48547: You can duplicate/multiply materials with tables
## Round ID: 

Locally tested
## Testmerges:
None

## Reproduction:
Build a table frame
Take 1 metal sheets and make 4 floor tiles
Use 1 floor tile to construct table
Deconstruct table for 1 sheet aka 4 tiles, now you got 4x the materials

Also works with metal rods:
Build table frame
Take 1 metal sheet and make 2 rods
Use 1 rod to make table
Deconstruct table for 1 sheet aka 2 rods, now you got 2x the materials

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

You can't multiply materials by crafting floor tiles or rods.
Also fixes some cases where mats from alloys would disappear upon merging stacks.

## Why It's Good For The Game

Infinite mats bad. Balance good. Fixes https://github.com/tgstation/tgstation/issues/48547

## Changelog
:cl:
fix: Fixes a material duplication bug.
/:cl:

### CODE REVIEW

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -350,7 +350,7 @@
 		src.amount += amount
 	if(custom_materials && custom_materials.len)
 		for(var/i in custom_materials)
-			custom_materials[getmaterialref(i)] = MINERAL_MATERIAL_AMOUNT * src.amount
+			custom_materials[i] = mats_per_stack * amount
```
Comment (kriskog) [inline code/game/objects/items/stacks/stack.dm]: src.amount != amount here

Comment (nemvar) [reply code/game/objects/items/stacks/stack.dm]: thanks

**Comment by nemvar** [comment]
Doesn't work on alloys. Makes the bug even worse.

**Comment by optimumtact** [comment]
Please consider this a friendly warning, your PR's must have a properly descriptive title and body content that explains the purpose of the PR.

Your current title and or body content does not meet requirements, so please update it.

**Comment by nemvar** [comment]
You used to be cool. :disappointed:

**Pull request merged**